			     BASH PATCH REPORT
			     =================

Bash-Release:	5.2
Patch-ID:	bash52-022

Bug-Reported-by:	srobertson@peratonlabs.com
Bug-Reference-ID:
Bug-Reference-URL:	https://lists.gnu.org/archive/html/bug-bash/2022-09/msg00049.html

Bug-Description:

It's possible for readline to try to zero out a line that's not null-
terminated, leading to a memory fault.

Patch (apply with `patch -p0'):

*** ../bash-5.2-patched/lib/readline/display.c	2022-04-05 10:47:31.000000000 -0400
--- lib/readline/display.c	2022-12-13 13:11:22.000000000 -0500
***************
*** 2684,2692 ****
  
    if (visible_line)
!     {
!       temp = visible_line;
!       while (*temp)
! 	*temp++ = '\0';
!     }
    rl_on_new_line ();
    forced_display++;
--- 2735,2740 ----
  
    if (visible_line)
!     memset (visible_line, 0, line_size);
! 
    rl_on_new_line ();
    forced_display++;

*** ../bash-5.2/patchlevel.h	2020-06-22 14:51:03.000000000 -0400
--- patchlevel.h	2020-10-01 11:01:28.000000000 -0400
***************
*** 26,30 ****
     looks for to find the patch level (for the sccs version string). */
  
! #define PATCHLEVEL 21
  
  #endif /* _PATCHLEVEL_H_ */
--- 26,30 ----
     looks for to find the patch level (for the sccs version string). */
  
! #define PATCHLEVEL 22
  
  #endif /* _PATCHLEVEL_H_ */
